doc/reference/dune-project/pin.rst

```diff
@@ -0,0 +1,43 @@
+pin
+---
+
+.. warning::
+
+   Dune Package Management is not final yet and the configuration options
+   are subject to change.
+
+
+Pins are package overrides used in the context of package management. They
+allow to fix a package at a specific version which is not affected by the
+package repositories selected.
+
+.. describe:: (pin ...)
+
+   .. versionadded:: 3.14
+
+   Define a package override.
+
+   .. describe:: (url <string>)
+
+      The URL of the package source.
+
+      This must be specified.
+
+   .. describe:: (package ...)
+
+      Defines which package is to be pinned.
+
+      This must be specified.
+
+      .. describe:: (name <string>)
+
+         The name of the package.
+
+         This must be specified.
+
+      .. describe:: (version <string>)
+
+         The version that the package should be assumed to be.
```

Comment:
Do we have a convention for `required` vs `optional` in the documentation? If not, I think we should think about some uniformity (not for this PR but keep it in a corner of our head).

:art:  I would suggest emphasising this, otherwise.
```suggestion
      This must **be specified**.

   .. describe:: (package ...)

      Defines which package is to be pinned.

      This must **be specified**.

      .. describe:: (name <string>)

         The name of the package.

         This must **be specified**.

      .. describe:: (version <string>)

         The version that the package should be assumed to be.
```

Reply:
> Do we have a convention for required vs optional in the documentation?

I think in most places it only mentions that a field is required and all fields are optional otherwise. (`rg required doc` vs `rg optional doc`) thus for consistency I would just write that fields are required where they are and nothing otherwise.